frontend/ui/widget/menu.lua

```diff
@@ -1107,6 +1108,26 @@ function Menu:updateItems(select_number)
     end)
 end
 
+-- merge TitleBar layout into self FocusManager layout
+function Menu:addTitleBarLayout()
+    local menu_item_layout_start_row = 1
+    local titlebars = {self.title_bar, self.outer_title_bar}
```

Comment:
I can't visualize the fact that we have 2 title bars in File manager :/ Is this really the case ?

Reply:
Yes. File Manager has TitleBar with Home and Plus buttons. Its menu widget has its own Titlebar without buttons but showing current directory.

Reply:
> Its menu widget has its own Titlebar without buttons but showing current directory.

Can you please show it? I see that the current directory is a subtitle of the TitleBar.
https://github.com/koreader/koreader/blob/d6ff9832431a2141b5f509c9ac5006204ee475fb/frontend/apps/filemanager/filemanager.lua#L104-L109

Reply:
Technically, FileManager makes its own title_bar, and use a FileChooser below.
FileChooser extends Menu, and Menu can have its own title_bar - but FileChooser extends Menu with providing `no_title=true`, which will have Menu builds its own title_bar, but actually _not_ inserting it into its main widget (which is indeed a bit confusing, and might be a old hack, but it's working :). Dunno if it inserts it into its focusmanager layout stuff.

So, it seems that even if we have the 2 title_bar built, the Menu/FileChooser one, not being in the displayed widget, won't be usable by FocusManager.
No issue with this merging trick? Should this trick be kept?

Reply:
@hius07 You are right. Current directory is a subtitle of Titlebar in FileManager. As @poire-z said, FileChooser set `no_title=true` to hide Menu's Titlebar.  The `Menu:mergeTitleBarIntoLayout` works well for these scenarios.

self.title_bar never be nil even no_title=true
self.outer_title_bar can be nil, well table length is 1 if it is nil, no extra if branch needed
https://github.com/koreader/koreader/blob/fa76b9d61527b8f96c1f9f1e6b64ea7f63e8fb4a/frontend/ui/widget/menu.lua#L1114

`generateVerticalLayout` returns empty layout table if no buttons in title bar, and the following for will be skipped.
https://github.com/koreader/koreader/blob/fa76b9d61527b8f96c1f9f1e6b64ea7f63e8fb4a/frontend/ui/widget/menu.lua#L1118-L1122